src/test_vector.rs

```diff
@@ -0,0 +1,229 @@
+
+/// An ECDSA P-256 private key suitable for decrypting inputs, used to generate
+/// test vectors and later to decrypt them.
+fn server_1_private_key() -> PrivateKey {
+    PrivateKey::from_base64(SERVER_1_PRIVATE_KEY).unwrap()
+}
+
+/// The public portion of [`server_1_private_key`].
+fn server_1_public_key() -> PublicKey {
+    PublicKey::from(&server_1_private_key())
+}
+
+/// An ECDSA P-256 private key suitable for decrypting inputs, used to generate
+/// test vectors and later to decrypt them.
+fn server_2_private_key() -> PrivateKey {
+    PrivateKey::from_base64(SERVER_2_PRIVATE_KEY).unwrap()
+}
+
+/// The public portion of [`server_2_private_key`].
+fn server_2_public_key() -> PublicKey {
+    PublicKey::from(&server_2_private_key())
+}
+
+/// A test vector of Prio inputs, serialized and encrypted in the Priov2/ENPA
+/// format, along with a reference sum.
+#[derive(Debug, Eq, PartialEq, Serialize, Deserialize)]
+pub struct EnpaTestVector<F> {
+    /// Base64 encoded private key for the "first" a.k.a. "PHA" server, which
+    /// may be used to decrypt `server_1_shares`.
+    pub server_1_private_key: String,
+    /// Base64 encoded private key for the non-"first" a.k.a. "facilitator"
+    /// server, which may be used to decrypt `server_2_shares`.
+    pub server_2_private_key: String,
+    /// Dimension (number of buckets) of the inputs
+    pub dimension: usize,
+    /// Encrypted shares of ENPA format inputs for the "first" a.k.a. "PHA"
+    /// server. The inner `Vec`s are encrypted bytes.
+    pub server_1_shares: Vec<Vec<u8>>,
+    /// Encrypted share of ENPA format inputs for the non-"first" a.k.a.
+    /// "facilitator" server.
+    pub server_2_shares: Vec<Vec<u8>>,
+    /// The sum over the inputs.
+    pub reference_sum: Vec<F>,
+    /// The version of the crate that generated this test vector
+    pub prio_crate_version: String,
+}
+
+impl<F: FieldElement> EnpaTestVector<F> {
+    /// Construct a test vector of `number_of_clients` inputs, each of which is a
+    /// `dimension`-dimension vector of random Boolean values encoded as `F`.
+    pub fn new(dimension: usize, number_of_clients: usize) -> Result<Self, TestVectorError> {
+        let mut client: Client<F> =
+            Client::new(dimension, server_1_public_key(), server_2_public_key())?;
+
+        let mut reference_sum = vec![F::zero(); dimension];
+        let mut server_1_shares = Vec::with_capacity(number_of_clients);
+        let mut server_2_shares = Vec::with_capacity(number_of_clients);
+
+        let mut rng = rand::thread_rng();
+
+        for _ in 0..number_of_clients {
+            // Generate a random vector of booleans
+            let data: Vec<F> = (0..dimension)
+                .map(|_| Ok(F::from(F::Integer::try_from(rng.gen_range(0..2))?)))
+                .collect::<Result<_, F::IntegerTryFromError>>()
+                .map_err(|e| TestVectorError::IntegerConversion(format!("{:?}", e)))?;
```

Comment:
- [ ] figure out if we can put a tighter bound on `FieldElement::IntegerTryFromError` to make this error conversion less ugly... or just do `F::Integer::try_from(...).unwrap()` since it's never really going to fail.